src/tests/testCacheManager.cc

```diff
@@ -66,3 +75,86 @@ testCacheManager::testRegister()
     CPPUNIT_ASSERT_EQUAL(1,(int)sentry->flags);
 }
 
+void
+testCacheManager::testParseUrl()
+{
+    auto *mgr = static_cast<CacheManagerInternals *>(CacheManager::GetInstance());
+    CPPUNIT_ASSERT(mgr != nullptr);
+
+    std::vector<AnyP::ProtocolType> validSchemes = {
+        AnyP::PROTO_CACHE_OBJECT,
+        AnyP::PROTO_HTTP,
+        AnyP::PROTO_HTTPS,
+        AnyP::PROTO_FTP
+    };
+
+    AnyP::Uri mgrUrl;
+    mgrUrl.host("localhost");
+    mgrUrl.port(3128);
+
+    const std::vector<const char *> validPathActions = {
+        // "", // technically valid, but not supported
+        "/",
+        "/menu",
+        // "/squid-internal-mgr", // technically valid, but not supported
+        "/squid-internal-mgr/",
+        "/squid-internal-mgr/menu"
+    };
+
+    const std::vector<const char *> validParams = {
+        "",
+        "?",
+        "?&",
+        "?&&&&&&&&&&&&",
+        "?foo=bar",
+        "?0123456789=bar",
+        "?foo=bar&",
+        "?foo=bar&&&&",
+        "?&foo=bar",
+        "?&&&&foo=bar",
+        "?&foo=bar&",
+        "?&&&&foo=bar&&&&",
+        "?foo=?_weird?~`:[]stuff&bar=okay&&&&&&"
+    };
```

Comment:
Since the parser attempts to parse parameter value lists, please add a few representative test cases for integer and non-integer lists, especially since that code has proven to be problematic already.

Please also add multi-parameter test cases. Ideally, they should be auto-generated (e.g., from hard-coded parameters) but that would require other improvements.

Reply:
Done in 77074c660.